test_projects/example/c_src/example/example_cnode.c

```diff
@@ -130,7 +137,13 @@ int main(int argc, char **argv) {
   int ei_fd = ei_accept_tmo(&ec, listen_fd, &conn, 5000);
   assert(ei_fd != ERL_ERROR);
 
-  int res = receive(ei_fd, node_name);
+  int res;
+  while(1) {
+    res = receive(ei_fd, node_name);
+    if(res != 0) {
+      break;
+    }
+  }
```

Comment:
This way we're going to repeat no matter which error happens. I think it's enough to stick to a single call to receive